lib/CssColors/Ecto/color.ex

```diff
@@ -57,4 +57,22 @@ defmodule CssColors.Ecto.Color do
     to_string CssColors.adjust(color, round_to, :alpha, fn(value, round_to) -> Float.round(value, round_to) end)
 
   end
+
+  @spec equal?(CssColors.color(), CssColors.color()) :: boolean()
+  def equal?(
+        %CssColors.RGB{red: red, green: green, blue: blue, alpha: alpha},
+        %CssColors.RGB{red: red, green: green, blue: blue, alpha: alpha}
+      ),
+      do: true
+
+  def equal?(
+        %CssColors.HSL{hue: hue, saturation: saturation, lightness: lightness, alpha: alpha},
+        %CssColors.HSL{hue: hue, saturation: saturation, lightness: lightness, alpha: alpha}
+      ),
+      do: true
+
+  def equal?(_, _), do: false
```

Comment:
Is `equal?` supposed to correspond to the equality check in the database? Got into interesting questions, such as:

* if `a` is the same color as `b`, but in different color schemes, should they be equal?
* should there be any fuzzyness? These values are floats, and could be very close to each other. In the DB, they are stored rounded to 4 digits.

Reply:
Thanks, that makes sense and didn't cross my mind!

The closest of the standard types in ecto is `DateTime` where the comparison uses `DateTime.compare` ignoring differences in how they are stored (primarily different timezones).

Also from the docs regarding the `equal?` function we get "Checks if two terms are semantically equal.", I read this as if `a` and `b` are the same color in different schemas they should be interpreted as equal.

Regarding fuzzyness, I would expect the comparison to match what's stored in the db (rounded).

I'll try to update with this!

Reply:
Looks great, added a comment. Would like to get rid of the catch all, the other is a suggestion.